src/sentry/static/sentry/app/views/settings/account/accountDetails.tsx

```diff
@@ -1,64 +1,53 @@
 import React from 'react';
 
-import {updateUser} from 'app/actionCreators/account';
+import {addErrorMessage, addSuccessMessage} from 'app/actionCreators/indicator';
 import AsyncView from 'app/views/asyncView';
 import AvatarChooser from 'app/components/avatarChooser';
 import Form from 'app/views/settings/components/forms/form';
 import JsonForm from 'app/views/settings/components/forms/jsonForm';
 import SettingsPageHeader from 'app/views/settings/components/settingsPageHeader';
 import accountDetailsFields from 'app/data/forms/accountDetails';
 import accountPreferencesFields from 'app/data/forms/accountPreferences';
+import {APIRequestMethod} from 'app/api';
+import {t} from 'app/locale';
 
 const ENDPOINT = '/users/me/';
 
 class AccountDetails extends AsyncView {
-  getEndpoints() {
+  getEndpoints(): ReturnType<AsyncView['getEndpoints']> {
     return [['user', ENDPOINT]];
   }
 
-  handleSubmitSuccess = user => {
-    updateUser(user);
-    this.setState({user});
-  };
-
-  handleSubmitError = (resp, model, id) => {};
-
   renderBody() {
     const {user} = this.state;
+    const {location} = this.props;
+
+    const formCommonProps: Partial<Form['props']> = {
+      apiEndpoint: ENDPOINT,
+      apiMethod: 'PUT' as APIRequestMethod,
+      onSubmitSuccess: () => {
+        addSuccessMessage(t('Successfully updated'));
+      },
+      onSubmitError: () => {
+        addErrorMessage(t('an error occurred while updating'));
+      },
```

Comment:
Ah so, `allowUndo` has a default success message, we should remove these and add `allowUndo` to the first `<Form>` instead.

```suggestion
      allowUndo: true,
```